langlib/lang.runtime/src/main/java/org/ballerinalang/langlib/runtime/Sleep.java

```diff
@@ -35,13 +35,24 @@
 public class Sleep {
 
     private static final int CORE_THREAD_POOL_SIZE = 1;
+    private static final BigDecimal LONG_MAX = new BigDecimal(Long.MAX_VALUE);
 
     private static final ScheduledExecutorService executor = Executors.newScheduledThreadPool(CORE_THREAD_POOL_SIZE);
 
     public static void sleep(Environment env, BDecimal delaySeconds) {
+        BigDecimal delayDecimal = delaySeconds.decimalValue();
+        if (delayDecimal.compareTo(BigDecimal.ZERO) <= 0) {
+            return;
+        }
+        delayDecimal = delayDecimal.multiply(new BigDecimal("1000.0"));
         Future balFuture = env.markAsync();
-        long delayMillis = (delaySeconds.decimalValue().multiply(new BigDecimal("1000.0"))).longValue();
-        executor.schedule(() -> balFuture.complete(null), delayMillis, TimeUnit.MILLISECONDS);
+        long delay;
+        if (delayDecimal.compareTo(LONG_MAX) > 0) {
+            delay = Long.MAX_VALUE;
```

Comment:
Any idea why this line is not covered. As per the test it's covered.

Reply:
The tests are covering the other cases. This one needs a larger `long` value as delay (`LONG_MAX`). Therefore, we can't add a test for this case.